### ISSUE TITLE
AndroidManifest not found with Gradle 2.2.0-alpha6

### ISSUE THREAD

**Issue by jcholin** [robolectric/robolectric#2581]
### Description

Hello,

I migrated to Android Studio Preview 6 and updated Gradle with 
`classpath 'com.android.tools.build:gradle:2.2.0-alpha6’`

Now I have an issue on build with Unit Tests : 

java.lang.RuntimeException: java.lang.RuntimeException: 
build/intermediates/manifests/full/mock/release/AndroidManifest.xml not found or not a file; it should point to your project's AndroidManifest.xml

I’ve checked the AndroidManifest.xml, no **full** folder found and it’s located now under **aapt** like this : 
build/intermediates/manifests/**aapt**/mock/release/AndroidManifest.xml
### Steps to Reproduce

On each build with the new version of Gradle
### Robolectric & Android Version

```
compileSdkVersion 24
buildToolsVersion "24.0.0"
testCompile 'org.robolectric:robolectric:3.0'
```

**Comment by jaredsburrows** [comment]
Try with `buildToolsVersion "24.0.1"` and `testCompile "org.robolectric:robolectric:3.1.1"`.

**Comment by jcholin** [comment]
Thanks, I tried, but now I get another issue on build which is : 

> You need to use a Theme.AppCompat theme (or descendant) with this activity

I already tried some tips like changing theme for App, Activivity from manifest, but nothing works.

**Comment by jaredsburrows** [comment]
@jcholin It has to be something with your setup as it works fine with my app. I could push up the changes to master if you would like.

**Comment by jaredsburrows** [comment]
@jcholin Are you still seeing this? Have you tried the latest `2.2.0-rc1`? I was able to reproduce it for an Android library project.

**Comment by venator85** [comment]
@jaredsburrows I had exactly the same problem, I put together a custom test runner that works:
https://gist.github.com/venator85/282df3677af9ecac56e5e4b91471cd8f

**Comment by digitalbuddha** [comment]
Thank you custom test runner works, is there any timetable for similar official fix?

**Comment by jongerrish** [comment]
Can you send a pull request with the fix? I'll happily accept it.

**Comment by digitalbuddha** [comment]
Yup! Will get it out tonight!

**Comment by jaredsburrows** [comment]
@digitalbuddha @jongerrish How is this a problem? All of my open source github projects use gradle android plugin `2.2.0` and gradle `3.1`. I do not see this problem at all.

**Comment by digitalbuddha** [comment]
I think its the new build tools. It seems like build tools 24.0.2 is putting the manifest in a different directory (called aapt now)

**Comment by jaredsburrows** [comment]
@digitalbuddha This doesnt make any sense. My project: https://github.com/jaredsburrows/android-gradle-java-app-template works fine.

**Comment by digitalbuddha** [comment]
Do you have flavors? How about multidex?  when you build your project, what directory is the manifest in?

**Comment by digitalbuddha** [comment]
@jaredsburrows I also noticed that my project uses APT while yours does not. Not sure if that makes a difference

**Comment by jaredsburrows** [comment]
No - avoid produceFlavor and use buildTypes if you can.
No - avoid multidex at all costs.
APT - should not matter at all, its only generated code in a different directory. Also, use the `annotationProccessor` that comes with `2.2.0`.

So I am guessing that robolectric needs to know about app's with product flavors.

**Comment by digitalbuddha** [comment]
OK a lot of that is opinion which should not be in this discussion. Our code base uses Flavors for amazon and google.  Multidex is on in debug builds where we are not proguarding. Flavors worked with last version of build tools.  I felt like this is a regression, if the project wants to drop support for flavors & multidex that should be in a major version upgrade.

**Comment by jaredsburrows** [comment]
@digitalbuddha Relax. We just got to the bottom of this. The only reason this does not work for you is `productFlavors`. Submit your PR.

**Comment by kingargyle** [comment]
This only seems to happen on Library Projects (i.e AAR) projects, the projects that are full APK projects for us aren't experiencing the issue.

**Comment by digitalbuddha** [comment]
Agreed. Should the pr be creating a new test runner for use in library
projects or should the pr make a runner that can be used in both cases? The
above gist works for me in both libraries as well as our main project. Let
me know whats best :-)

**Comment by jongerrish** [comment]
Can you add to the existing test runner please?

We're going to experiment with a gradle plugin to provide these file
locations to the runner via system properties, if it works out we will just
port the logic included from the test runner, so adding it there will be
helpful.

**Comment by jaredsburrows** [comment]
@venator85 Thanks for posting this! This is good for library projects. Everything seems fine for regular android projects.

**Comment by peter-budo** [comment]
@jaredsburrows any idea when new version with this fix would be released?

**Comment by jaredsburrows** [comment]
@peter-budo I am not a collaborator on this project. You should ask @jongerrish or @xian.

**Comment by peter-budo** [comment]
Sorry @jaredsburrows 😉 
@jongerrish @xian  any update on this being released any time soon?

**Comment by xiphirx** [comment]
@peter-budo This should be fixed once my PR lands: https://github.com/robolectric/robolectric/issues/2673
